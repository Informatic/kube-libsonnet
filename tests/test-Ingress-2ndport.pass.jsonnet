local bitnami = import "../bitnami.libsonnet";
local kube = import "../kube.libsonnet";

local stack = {
  name:: "test-Ingress-fail",
  pod: kube.Pod($.name + "-pod") {
    spec+: {
      containers_+: {
        foo_cont: kube.Container($.name) {
          image: "nginx:1.12",
          ports_+: {
            http: { containerPort: 80 },
            metrics: { containerPort: 9099 },
          },
        },
      },
    },
  },
  deploy: kube.Deployment($.name + "-deploy") {
    local this = self,
    spec+: {
      template+: {
        spec+: $.pod.spec {
        },
      },
    },
  },
  service: kube.Service($.name + "-svc") {
    local this = self,
    target_pod: $.deploy.spec.template,
    name_port+:: {
      default_port:: this.target_pod.spec.containers[0].ports[1],
    },
  },
  ingress: bitnami.Ingress($.name + "-ingress") {
    host: "foo.g.dev.bitnami.net",
    target_svc: $.service,
  },
};

stack {
  assert (stack.ingress.spec.rules[0].http.paths[0].backend.service.port == { name: "metrics" }),
}
