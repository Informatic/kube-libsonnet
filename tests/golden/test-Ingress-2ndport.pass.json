{
   "deploy": {
      "apiVersion": "apps/v1",
      "kind": "Deployment",
      "metadata": {
         "annotations": { },
         "labels": {
            "name": "test-Ingress-fail-deploy"
         },
         "name": "test-Ingress-fail-deploy"
      },
      "spec": {
         "minReadySeconds": 30,
         "replicas": 1,
         "revisionHistoryLimit": 10,
         "selector": {
            "matchLabels": {
               "name": "test-Ingress-fail-deploy"
            }
         },
         "strategy": {
            "rollingUpdate": {
               "maxSurge": "25%",
               "maxUnavailable": "25%"
            },
            "type": "RollingUpdate"
         },
         "template": {
            "metadata": {
               "annotations": { },
               "labels": {
                  "name": "test-Ingress-fail-deploy"
               }
            },
            "spec": {
               "containers": [
                  {
                     "args": [ ],
                     "env": [ ],
                     "image": "nginx:1.12",
                     "imagePullPolicy": "IfNotPresent",
                     "name": "test-Ingress-fail",
                     "ports": [
                        {
                           "containerPort": 80,
                           "name": "http"
                        },
                        {
                           "containerPort": 9099,
                           "name": "metrics"
                        }
                     ],
                     "stdin": false,
                     "tty": false,
                     "volumeMounts": [ ]
                  }
               ],
               "imagePullSecrets": [ ],
               "initContainers": [ ],
               "terminationGracePeriodSeconds": 30,
               "volumes": [ ]
            }
         }
      }
   },
   "ingress": {
      "apiVersion": "networking.k8s.io/v1",
      "kind": "Ingress",
      "metadata": {
         "annotations": {
            "cert-manager.io/cluster-issuer": "letsencrypt-prod-dns"
         },
         "labels": {
            "name": "test-Ingress-fail-ingress"
         },
         "name": "test-Ingress-fail-ingress"
      },
      "spec": {
         "rules": [
            {
               "host": "foo.g.dev.bitnami.net",
               "http": {
                  "paths": [
                     {
                        "backend": {
                           "service": {
                              "name": "test-Ingress-fail-svc",
                              "port": {
                                 "name": "metrics"
                              }
                           }
                        },
                        "path": "/",
                        "pathType": "ImplementationSpecific"
                     }
                  ]
               }
            }
         ],
         "tls": [
            {
               "hosts": [
                  "foo.g.dev.bitnami.net"
               ],
               "secretName": "test-Ingress-fail-ingress-cert"
            }
         ]
      }
   },
   "pod": {
      "apiVersion": "v1",
      "kind": "Pod",
      "metadata": {
         "annotations": { },
         "labels": {
            "name": "test-Ingress-fail-pod"
         },
         "name": "test-Ingress-fail-pod"
      },
      "spec": {
         "containers": [
            {
               "args": [ ],
               "env": [ ],
               "image": "nginx:1.12",
               "imagePullPolicy": "IfNotPresent",
               "name": "test-Ingress-fail",
               "ports": [
                  {
                     "containerPort": 80,
                     "name": "http"
                  },
                  {
                     "containerPort": 9099,
                     "name": "metrics"
                  }
               ],
               "stdin": false,
               "tty": false,
               "volumeMounts": [ ]
            }
         ],
         "imagePullSecrets": [ ],
         "initContainers": [ ],
         "terminationGracePeriodSeconds": 30,
         "volumes": [ ]
      }
   },
   "service": {
      "apiVersion": "v1",
      "kind": "Service",
      "metadata": {
         "annotations": { },
         "labels": {
            "name": "test-Ingress-fail-svc"
         },
         "name": "test-Ingress-fail-svc"
      },
      "spec": {
         "ports": [
            {
               "name": "http",
               "port": 80,
               "targetPort": 80
            }
         ],
         "selector": {
            "name": "test-Ingress-fail-deploy"
         },
         "type": "ClusterIP"
      }
   }
}
